### ISSUE TITLE
Question regarding OneToMany relationships and querying them

### ISSUE THREAD

**Issue by pushrbx** [KristianOellegaard/django-hvad#305]
I was looking for some example in the docs which would show me how a query looks like when you have two models in one to many relationship and both of them have translatable fields but I couldn't find any. Could you please show me an example how a query would look like in such a case?

My models:

    class Vocabulary(TranslatableModel):
        alias = models.CharField(max_length=255, blank=True, null=True)
        required = models.BooleanField(default=False, null=False)
        multiple = models.BooleanField(default=False, null=False)
        updated = models.DateTimeField(auto_now=True)
        created = models.DateTimeField(auto_now_add=True)

        translations = TranslatedFields(
            description=models.TextField(blank=True, null=True),
            title=models.CharField(max_length=255),
        )

    class Term(TranslatableModel):
        slug = models.CharField(max_length=255)
        updated = models.DateTimeField(auto_now=True)
        created = models.DateTimeField(auto_now_add=True)
        vocabulary = models.ForeignKey(Vocabulary, related_name="terms")

        translations = TranslatedFields(
            description=models.TextField(blank=True, null=True),
            title=models.CharField(max_length=255),
        )

I tried this query, but it gives me the WrongManager exception:
`Term.objects.language('en').filter(title="mystery", vocabulary__title="genres")[:1])`

Basically my goal with this query is to get all the Terms in English where the vocabulary title is "genres" in English.

Thanks in advance.

**Comment by spectras** [comment]
Hello,

Deep filters are not supported right now, tough that's something I really want to have for hvad2.
In the meantime, it can be emulated by filtering through the translations manually, like this:

```python
Term.objects.language('en').filter(title="mystery",
                                   vocabulary__translations__language_code='en',
                                   vocabulary__translations__title="genres")
```
Filtering on the language is mandatory as we are not using hvad's automated language support here. If we did not filter on the language, it would return all terms that have a vocabulary titled `genres` in any language.

As this workaround is commonly used, we'll make sure to have a proper deprecation path for it once deep filters are supported.

**Comment by pushrbx** [comment]
One more thing then on this topic:
How can I filter on untranslated fields without having the WrongManager exception?
Let's say I'd like to have a Term which has an untranslated title which is "cars", a translated english field which is also "cars", and I'd like to filter on the untranslated "cars". 

Also I think it would be nice to include this small piece of information in the documentation.

**Comment by spectras** [comment]
I'm not sure I understand what you mean with untranslated title? `title` is a translated field in the models you posted. Can you show what you are trying to do that causes the exception?

**Comment by pushrbx** [comment]
Might be my understanding of the library is wrong. I thought that I still have my normal "title" field on my model, and the translated "title" fields are subsequent bits on the model. Like if you would have multiple fields with the same name but for different languages. My question is based on this understanding.

**Comment by pushrbx** [comment]
Just for reference the query I'm trying to do:
`Term.objects.filter(title="cars")[:1]`

**Comment by spectras** [comment]
Oh, I understand. Well, there is no concept of "untranslated translated field". In hvad, all languages are equal, there is no “main”, “default” or “untranslated” language. In fact, it is invalid for a model to not have at least one translation.

So you have to pick a language to use for filtering. Common values could be `en`, or `settings.LANGUAGE` to use your website's default language, or simply leave the language empty (`Term.objects.language()`) to use the user's browser settings (requires using Django-provided LanguageMiddleware).

On the technical side, a `TranslatableModel` will be materialized into two tables. The shared table has all untranslated fields, and the translations table has all translated fields, plus autogenerated language_code and foreign key to shared table.

**Comment by pushrbx** [comment]
I see now. Thanks. Now it's all clear for me.

**Comment by spectras** [comment]
Great, feel free to open another ticket should you have another question.
